### PR TITLE
chore(profiling): add native ID to user threads

### PR DESCRIPTION
**Pull request by P403n1x87**
Patch Thread._bootstrap to retrieve and store the native thread ID (the TID) on Linux. This might come handy for determining whether a call stack is on CPU or not.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: tests/profiling/collector/test_threading.py [hunk 1]
```diff
@@ -312,3 +313,40 @@ def test_lock_acquire_release_speed_patched(benchmark, pct):
...
+    from os import getpid
+    from threading import Thread
+    from threading import _MainThread
+    from threading import current_thread
+    from time import sleep
+
+    main = current_thread()
+    assert isinstance(main, _MainThread)
+    # We expect the main thread to have the same ID as the PID
+    assert main.native_id == getpid(), (main.native_id, getpid())
```
Comment (sanchda) [inline tests/profiling/collector/test_threading.py:334]: +1 to marking this test Linux-only, as this assumption in particular obviously breaks on other platforms (even macos).

File: tests/profiling/collector/test_threading.py [hunk 2]
```diff
@@ -312,3 +313,40 @@ def test_lock_acquire_release_speed_patched(benchmark, pct):
...
+
+    t = Thread(target=lambda: None)
+    t.start()
+
+    for _ in range(10):
+        try:
+            # The TID should be higher than the PID, but not too high
```
Comment (sanchda) [inline tests/profiling/collector/test_threading.py:341]: I think it's a good idea to sanity-test pid/tid integrity, but this assumption will break when
* PIDs are just about to wraparound
* The underlying PID namespace has a lot of activity, especially if the current task gets de-scheduled

Neither of these considerations is relevant for CI, so I think it's perfectly fine to take this section--just wanted to callout potential sources of flakiness.

**Review by sanchda — APPROVED** [review]
Made some purely editorial comments.  This looks great!  It really makes me want to move more of the Profiler implementation into native code so we can rationalize the behavior through the native profiler.  ❤️

File: tests/profiling/collector/test_threading.py (hunk 2 above)
Comment (P403n1x87) [reply tests/profiling/collector/test_threading.py:341]: Indeed. I've thought about the first point you made and assumed it should be extremely rare to hit the wraparound boundary in CI. So hopefully we're good 🤞

**Pull request merged**